### ISSUE TITLE
cockroach reports "unable to allocate a target store; no candidates available"  when running cockroach locally 

### ISSUE THREAD

**Issue by jerry0715** [cockroachdb/cockroach#2421]
Hi,

Here is my steps followed by "Running Cockroach Locally" section after pull docker image.
-  (1) pull docker image
```{r, engine='bash', count_lines}
docker pull cockroachdb/cockroach
```

-  (2) start docker 
```{r, engine='bash', count_lines}
docker run -t -i -p 8080:8080 cockroachdb/cockroach shell
```

- (3) init & start cockroach inside docker
```{r, engine='bash', count_lines}
DIR=$(mktemp -d /tmp/dbXXXXXX)
./cockroach cert create-ca
./cockroach cert create-node 127.0.0.1 localhost $(hostname)
./cockroach cert create-client root
./cockroach init --stores=ssd=$DIR
./cockroach start --stores=ssd="$DIR" --gossip=self= &
```

Is there any step I missed or did incorrectly? Thanks.

operational log:
----
/cockroach #
/cockroach # DIR=$(mktemp -d /tmp/dbXXXXXX)
/cockroach # ./cockroach cert create-ca
/cockroach # ./cockroach cert create-node 127.0.0.1 localhost $(hostname)
/cockroach # ./cockroach cert create-client root
/cockroach # ./cockroach init --stores=ssd=$DIR
I0909 05:21:43.552349 42 server/context.go:172  initialized 1 storage engine(s)
I0909 05:21:43.552550 42 storage/engine/rocksdb.go:99  opening rocksdb instance at "/tmp/dbuJtVRA"
I0909 05:21:43.560354 42 storage/replica_command.go:1045  range 1: new leader lease replica 1:1 00:00:00.000 +1441776104.559s
I0909 05:21:43.562422 42 storage/engine/rocksdb.go:124  closing rocksdb instance at "/tmp/dbuJtVRA"
I0909 05:21:43.563106 42 cli/start.go:78  cockroach cluster 4cc2cd74-5dc9-4c9a-aa37-24779ac9ae35 has been initialized
/cockroach # ./cockroach start --stores=ssd="$DIR" --gossip=self= &
/cockroach # I0909 05:21:58.343760 48 cli/start.go:108  build Vers: go1.5
I0909 05:21:58.344150 48 cli/start.go:109  build Tag:  alpha-4710-g4f32087
I0909 05:21:58.344488 48 cli/start.go:110  build Time: 2015/09/09 04:56:38
I0909 05:21:58.344836 48 cli/start.go:111  build Deps: github.com/biogo/store:e1f74b3c58befe661feed7fa4cf52436de753128 github.com/cockroachdb/c-lz4:6e71f140a365017bbe0904710007f8725fd3f809 github.com/cockroachdb/c-protobuf:6a18bfcdd5169966cd1235edb71d749c950216e8 github.com/cockroachdb/c-rocksdb:7fc876fe79b96de0e25069c9ae27e6444637bd54 github.com/cockroachdb/c-snappy:618733f9e5bab8463b9049117a335a7a1bfc9fd5 github.com/cockroachdb/cockroach:4f32087e1033c52b5efa244c9b6fc5a8aeb66815 github.com/coreos/etcd:ae2b43b588fa39fb99598d03a7a554549595bddd github.com/cpuguy83/go-md2man:71acacd42f85e5e82f70a55327789582a5200a90 github.com/elazarl/go-bindata-assetfs:d5cac425555ca5cf00694df246e04f05e6a55150 github.com/gogo/protobuf:932b70afa8b0bf4a8e167fdf0c3367cebba45903 github.com/google/btree:cc6329d4279e3f025a53a83c397d2339b5705c45 github.com/inconshreveable/mousetrap:76626ae9c91c4f2a10f34cad8ce83ea42c93bb75 github.com/julienschmidt/httprouter:6aacfd5ab513e34f7e64ea9627ab9670371b34e7 github.com/montanaflynn/stats:b29c7dc0aff48386203e2508b144a3630c2d726c github.com/olekukonko/tablewriter:b9346ac189c55dd419f85c7ad2cd56f810bf19d6 github.com/peterh/liner:c754da6f2d91ef30ddb6c975d2dbe7696eec4fbc github.com/russross/blackfriday:8cec3a854e68dba10faabbe31c089abf4a3e57a6 github.com/shurcooL/sanitized_anchor_name:244f5ac324cb97e1987ef901a0081a77bfd8e845 github.com/spf13/cobra:e4993076d845b7127c760e2e57a4984c166c6c05 github.com/spf13/pflag:8e7dc108ab3a1ab6ce6d922bbaff5657b88e8e49 golang.org/x/crypto:81bf7719a6b7ce9b665598222362b50122dfc13b golang.org/x/net:7654728e381988afd88e58cabfd6363a5ea91810 golang.org/x/text:505f8b49cc14d790314b7535959a10b87b9161c7 gopkg.in/yaml.v1:9f9df34309c04878acc86042b16630b0f696e1de
I0909 05:21:58.345295 48 server/context.go:172  initialized 1 storage engine(s)
I0909 05:21:58.345588 48 cli/start.go:133  starting cockroach cluster
I0909 05:21:58.348942 48 storage/engine/rocksdb.go:99  opening rocksdb instance at "/tmp/dbuJtVRA"
I0909 05:21:58.355449 48 server/node.go:307  initialized store store=1:1 ([ssd]=/tmp/dbuJtVRA): {Capacity:21100904448 Available:15476539392 RangeCount:0}
I0909 05:21:58.355856 48 server/node.go:214  new node allocated ID 1
I0909 05:21:58.356381 48 gossip/gossip.go:163  gossiping node descriptor node_id:1 address:<network_field:"tcp" address_field:"f41dc2b0f14d:8080" > attrs:<>
I0909 05:21:58.356633 48 server/node.go:397  connecting to gossip network to verify cluster ID...
I0909 05:21:58.356963 48 server/node.go:414  node connected via gossip and verified as part of cluster "4cc2cd74-5dc9-4c9a-aa37-24779ac9ae35"
I0909 05:21:58.357310 48 server/node.go:272  Started node with [[ssd]=/tmp/dbuJtVRA] engine(s) and attributes []
I0909 05:21:58.357585 48 server/server.go:196  starting https server at f41dc2b0f14d:8080
E0909 05:22:03.357482 48 storage/queue.go:311  failure processing replica range=1 [""-"\xff\xff") from replicate queue: storage/allocator.go:126: unable to allocate a target store; no candidates available
E0909 05:22:08.357791 48 storage/queue.go:311  failure processing replica range=1 [""-"\xff\xff") from replicate queue: storage/allocator.go:126: unable to allocate a target store; no candidates available

/cockroach # E0909 05:22:13.357737 48 storage/queue.go:311  failure processing replica range=1 [""-"\xff\xff") from replicate queue: storage/allocator.go:126: unable to allocate a target store; no candidates available
E0909 05:22:18.365354 48 storage/queue.go:311  failure processing replica range=1 [""-"\xff\xff") from replicate queue: storage/allocator.go:126: unable to allocate a target store; no candidates available
E0909 05:22:23.358651 48 storage/queue.go:311  failure processing replica range=1 [""-"\xff\xff") from replicate queue: storage/allocator.go:126: unable to allocate a target store; no candidates available
E0909 05:22:28.358978 48 storage/queue.go:311  failure processing replica range=1 [""-"\xff\xff") from replicate queue: storage/allocator.go:126: unable to allocate a target store; no candidates available
E0909 05:22:33.359669 48 storage/queue.go:311  failure processing replica range=1 [""-"\xff\xff") from replicate queue: storage/allocator.go:126: unable to allocate a target store; no candidates available
E0909 05:22:38.360032 48 storage/queue.go:311  failure processing replica range=1 [""-"\xff\xff") from replicate queue: storage/allocator.go:126: unable to allocate a target store; no candidates available
E0909 05:22:43.363268 48 storage/queue.go:311  failure processing replica range=1 [""-"\xff\xff") from replicate queue: storage/allocator.go:126: unable to allocate a target store; no candidates available
E0909 05:22:48.360567 48 storage/queue.go:311  failure processing replica range=1 [""-"\xff\xff") from replicate queue: storage/allocator.go:126: unable to allocate a target store; no candidates available

/cockroach #
/cockroach # ps -elf
PID   USER     COMMAND
    1 root     {cockroach.sh} /bin/sh /cockroach/cockroach.sh shell
    6 root     /bin/sh
   48 root     ./cockroach start --stores=ssd=/tmp/dbuJtVRA --gossip=self=
   56 root     ps -elf
/cockroach # E0909 05:22:53.361973 48 storage/queue.go:311  failure processing replica range=1 [""-"\xff\xff") from replicate queue: storage/allocator.go:126: unable to allocate a target store; no candidates available
kill -9 48E0909 05:22:58.362216 48 storage/queue.go:311  failure processing replica range=1 [""-"\xff\xff") from replicate queue: storage/allocator.go:126: unable to allocate a target store; no candidates available

/cockroach #
[1]+  Killed                     ./cockroach start --stores=ssd="${DIR}" --gossip=self=
/cockroach #

/cockroach # find .
.
./certs
./certs/ca.key
./certs/ca.crt
./certs/node.client.crt
./certs/root.client.crt
./certs/node.client.key
./certs/node.server.crt
./certs/node.server.key
./certs/root.client.key
./cockroach
./cockroach.sh

cockroach version
--
/cockroach # ./cockroach version
Build Vers:  go1.5
Build Tag:   alpha-4710-g4f32087
Build Time:  2015/09/09 04:56:38
Build Deps:
  github.com/biogo/store                     e1f74b3c58befe661feed7fa4cf52436de753128
  github.com/cockroachdb/c-lz4               6e71f140a365017bbe0904710007f8725fd3f809
  github.com/cockroachdb/c-protobuf          6a18bfcdd5169966cd1235edb71d749c950216e8
  github.com/cockroachdb/c-rocksdb           7fc876fe79b96de0e25069c9ae27e6444637bd54
  github.com/cockroachdb/c-snappy            618733f9e5bab8463b9049117a335a7a1bfc9fd5
  github.com/cockroachdb/cockroach           4f32087e1033c52b5efa244c9b6fc5a8aeb66815
  github.com/coreos/etcd                     ae2b43b588fa39fb99598d03a7a554549595bddd
  github.com/cpuguy83/go-md2man              71acacd42f85e5e82f70a55327789582a5200a90
  github.com/elazarl/go-bindata-assetfs      d5cac425555ca5cf00694df246e04f05e6a55150
  github.com/gogo/protobuf                   932b70afa8b0bf4a8e167fdf0c3367cebba45903
  github.com/google/btree                    cc6329d4279e3f025a53a83c397d2339b5705c45
  github.com/inconshreveable/mousetrap       76626ae9c91c4f2a10f34cad8ce83ea42c93bb75
  github.com/julienschmidt/httprouter        6aacfd5ab513e34f7e64ea9627ab9670371b34e7
  github.com/montanaflynn/stats              b29c7dc0aff48386203e2508b144a3630c2d726c
  github.com/olekukonko/tablewriter          b9346ac189c55dd419f85c7ad2cd56f810bf19d6
  github.com/peterh/liner                    c754da6f2d91ef30ddb6c975d2dbe7696eec4fbc
  github.com/russross/blackfriday            8cec3a854e68dba10faabbe31c089abf4a3e57a6
  github.com/shurcooL/sanitized_anchor_name  244f5ac324cb97e1987ef901a0081a77bfd8e845
  github.com/spf13/cobra                     e4993076d845b7127c760e2e57a4984c166c6c05
  github.com/spf13/pflag                     8e7dc108ab3a1ab6ce6d922bbaff5657b88e8e49
  golang.org/x/crypto                        81bf7719a6b7ce9b665598222362b50122dfc13b
  golang.org/x/net                           7654728e381988afd88e58cabfd6363a5ea91810
  golang.org/x/text                          505f8b49cc14d790314b7535959a10b87b9161c7
  gopkg.in/yaml.v1                           9f9df34309c04878acc86042b16630b0f696e1de

**Comment by jerry0715** [comment]
Then execute a client command with error

/cockroach # ./cockroach kv put a 1 b 2 c 3 d 4
W0909 05:55:36.014154 42 client/rpc_sender.go:92  failed to send RPC request Server.Batch: client is unhealthy
W0909 05:55:36.015929 42 rpc/client.go:227  x509: certificate is valid for 127.0.0.1, not 172.17.0.3
I0909 05:55:36.022752 33 server.go:1934  http: TLS handshake error from 172.17.0.3:39996: remote error: bad certificate
W0909 05:55:36.069722 42 client/rpc_sender.go:92  failed to send RPC request Server.Batch: client is unhealthy
W0909 05:55:36.168871 42 client/rpc_sender.go:92  failed to send RPC request Server.Batch: client is unhealthy
W0909 05:55:36.377722 42 client/rpc_sender.go:92  failed to send RPC request Server.Batch: client is unhealthy
W0909 05:55:36.826288 42 client/rpc_sender.go:92  failed to send RPC request Server.Batch: client is unhealthy
W0909 05:55:36.912981 42 rpc/client.go:227  x509: certificate is valid for 127.0.0.1, not 172.17.0.3
I0909 05:55:36.916717 33 server.go:1934  http: TLS handshake error from 172.17.0.3:39997: remote error: bad certificate
W0909 05:55:37.564041 42 client/rpc_sender.go:92  failed to send RPC request Server.Batch: client is unhealthy
put failed: failed to send RPC request Server.Batch: client is unhealthy
/cockroach # E0909 05:55:38.445562 33 storage/queue.go:311  failure processing replica range=1 [""-"\xff\xff") from replicate queue: storage/allocator.go:126: unable to allocate a target store; no candidates available

**Comment by tamird** [comment]
> W0909 05:55:36.015929 42 rpc/client.go:227 x509: certificate is valid for 127.0.0.1, not 172.17.0.3
I0909 05:55:36.022752 33 server.go:1934 http: TLS handshake error from 172.17.0.3:39996: remote error: bad certificate

This is the problem; looks like you generated certs for 127.0.0.1:
> ./cockroach cert create-node 127.0.0.1 localhost $(hostname)

But are connecting to a local server on your resolved localhost address (172.17.0.3):
> ./cockroach kv put a 1 b 2 c 3 d 4

Can you try connecting to 127.0.0.1 explicitly?
> ./cockroach kv put a 1 b 2 c 3 d 4 --addr 127.0.0.1:8080

**Comment by jerry0715** [comment]
Hi,

My environment is Ubuntu (3.13.0-24-generic #47-Ubuntu, x86_64) with docker (1.8.1) and there are two IPs, localhost is 127.0.0.1 and eth0 is 172.17.0.x. I tried to use both 127.0.0.1 and 172.17.0.x. It still has a problem with "client is unhealthy."  from latest image (cockroachdb/cockroach   latest  1b2c1b79a4ca )

Please correct me if I am wrong. Or should multiple host name or IP addresses on same machine be supported?

trace:
   0 DIR=$(mktemp -d /tmp/dbXXXXXX)
   1 ./cockroach cert create-ca
   2 ./cockroach cert create-node 127.0.0.1
   3 find .
   4 ./cockroach cert create-client root
   5 find .
   6 uname -a
   7 ./cockroach init --stores=ssd=$DIR
   8 ./cockroach start --addr 127.0.0.1:8080 --stores=ssd="$DIR" --gossip=self= &
   9 ./cockroach kv put a 1 b 2 c 3 d 4 --addr 127.0.0.1:8080

Then I tried to use 172.17.0.x for all steps and also errors are reported.
